File: movie-domain/src/main/java/com/movie/domain/entity/Theater.java
package com.movie.domain.entity;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;
import lombok.Getter;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "theater")
public class Theater extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToMany(mappedBy = "theater", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Seat> seats = new ArrayList<>();

    public Theater(String name) {
        this.name = name;
    }

    public void updateName(String name) {
        this.name = name;
    }

    public void addSeat(Seat seat) {
        this.seats.add(seat);
    }

    public void initializeSeats() {
        char[] rows = {'A', 'B', 'C', 'D', 'E'};
        for (char row : rows) {
            for (int col = 1; col <= 5; col++) {
                String seatNumber = row + String.valueOf(col);
                addSeat(new Seat(seatNumber, this));
            }
        }
    }
}

File: core/src/main/kotlin/kr/spartacodingclub/cinema/core/domain/Theater.kt
package kr.spartacodingclub.cinema.core.domain

data class Theater(
    val id: Long,
    val name: String,
    val seats: List<Seat> = generateSeats()
) {
    companion object {
        private fun generateSeats(): List<Seat> {
            return ('A'..'E').flatMap { row ->
                (1..5).map { col ->
                    Seat("$row$col")
                }
            }
        }
    }
}
